src/meshapi/views/forms.py

```diff
@@ -209,6 +209,34 @@ def get_next_free_nn() -> int:
     if free_nn < NETWORK_NUMBER_MIN or free_nn > NETWORK_NUMBER_MAX:
         raise ValueError(f"Invalid NN: {free_nn}")
 
+    # The number we are about to assign should not be connected to any existing installs as
+    # an NN. Again, the above logic should do this, but we REALLY care about this not happening
+    already_in_use_nn_qs = Install.objects.filter(network_number=free_nn)
+    if len(already_in_use_nn_qs):
+        raise ValueError(
+            f"Invalid NN: {free_nn} is already in use for "
+            f"install number {already_in_use_nn_qs.first().install_number}"
+        )
+
+    # If we are re-assigning a number from another install, mark it with NN Assigned to indicate
+    # that this has happened
+    nn_donor_install: Optional[Install] = Install.objects.select_for_update().filter(install_number=free_nn).first()
+    if nn_donor_install:
+        # Double check that if we are re-assigning something that has been used before that it is
+        # definitely unused. The logic above should do that, but this is so important that for
+        # safety that we should double-check
+        if (
+            nn_donor_install.install_status != Install.InstallStatus.REQUEST_RECEIVED
+            or nn_donor_install.network_number is not None
+        ):
+            raise ValueError(
+                f"Invalid NN: {free_nn} has an install associated that "
+                f"looks active (#{nn_donor_install.install_number})"
+            )
+
+        nn_donor_install.install_status = Install.InstallStatus.NN_ASSIGNED
+        nn_donor_install.save()
```

Comment:
Why not simply leave the NN field blank and use another status like `CLOSED` or something? This information doesn't seem that useful, and this seems completely irrelevant for installs going forward.